### DISCUSSION TITLE
REST API v2 returns 404 for all endpoints

### DISCUSSION THREAD

**Discussion by tomaszmwojcik** [drakkan/sftpgo#286] [Q&A] [answered]
Hello,
I'm new to this project so it might be that I'm missing something obvious but I've noticed that recently REST API has been revamped recently so I thought double checking here could be a good idea.

I'm running the latest released version of the server via docker using:
`docker run --name sftpgo-test -v sftpgo:/srv/sftpgo -p 127.0.0.1:8080:8080 -p 2022:2022 -d "drakkan/sftpgo:v1.2.2"`
the web interface is accessible, e.g. I can open `http://localhost:8080/web/users`, however REST API doesn't seem to work
```
curl --location --request GET 'http://localhost:8080/api/v2/token' \
--header 'Accept: application/json' \
--header 'Authorization: Basic YWRtaW46cGFzc3dvcmQ='
``` 
returns 404 with `{"message":"Not Found"}`. I've got the same response from `/api/v2/version`
The URL(`/api/v2/token`) and login/password(`admin/password`) I'm using have been taken from https://github.com/drakkan/sftpgo/blob/master/docs/rest-api.md 
I'm assuming REST API uses the same port as the web interface and doesn't have to be explicitly enabled as the config and docker sections of docs don't mention that. 
Any idea what is wrong there? Thanks

**Comment by drakkan** [comment, marked as answer]
Hi,

you are using `drakkan/sftpgo:v1.2.2`, if you want to test REST API V2 (this is very appreciated), you have to use the `edge` image variant (current git master) that will become v2.0.0 in few weeks, thank you!

**Reply by tomaszmwojcik** [reply]
Thanks a lot for the quick response. Indeed, API v1 is working. Unfortunately before asking the question I checked only `/api/v1/token` endpoint which is not present in it so I thought both API versions are down. I should also have switched to v1 tag in the documentation but I assumed that master is the last released version. 
Thanks again for your help and this project, looks like it's exactly what I was looking for!

**Reply by drakkan** [reply]
No problem, 

if you are doing a new installation I recommend starting from git master, so you will help test the next version and avoid a migration in a few weeks, thanks

**Reply by tomaszmwojcik** [reply]
Sure, I'll give it a try. Thanks
